pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -321,13 +326,153 @@ func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEventWithHin
+func (pl *VolumeRestrictions) isSchedulableAfterNodeAdded(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	_, newNode, err := util.As[*v1.Node](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterNodeAdded: %w", err)
+	}
+
+	pvcs, err := pl.readWriteOncePodPVCsForPod(pod, false)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.V(5).Info("no PVC for the Pod is found, this Pod won't be schedulable until PVC is created", "pod", klog.KObj(pod), "err", err)
+			return framework.QueueSkip, nil
+		}
+		return framework.Queue, err
+	}
+
+	for pvc := range pvcs {
+		key := framework.GetNamespacedName(pod.Namespace, pvc)
+		if pl.sharedLister.StorageInfos().IsPVCUsedByPods(key) {
+			logger.V(5).Info("The Pod uses a PVC with ReadWriteOncePod access mode that is occupied by another Pod.", "pod", klog.KObj(pod))
+			return framework.QueueSkip, nil
+		}
+	}
+
+	logger.V(5).Info("node was added", "pod", klog.KObj(pod), "node", klog.KObj(newNode))
+	return framework.Queue, nil
+}
+
+// isSchedulableAfterPersistentVolumeClaimChange is invoked whenever a PersistentVolumeClaim changed, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldPersistentVolumeClaim, newPersistentVolumeClaim, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterPersistentVolumeClaimChange: %w", err)
+	}
+
+	if newPersistentVolumeClaim.Namespace != pod.Namespace {
+		return framework.QueueSkip, nil
+	}
+
+	pvcs := sets.New[string]()
+	readWriteOncePodPVC := sets.New[string]()
+
+	for _, volume := range pod.Spec.Volumes {
+		if volume.PersistentVolumeClaim == nil {
+			continue
+		}
+
+		pvc, err := pl.pvcLister.PersistentVolumeClaims(pod.Namespace).Get(volume.PersistentVolumeClaim.ClaimName)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				logger.V(5).Info("no PVC for the Pod is found, this Pod won't be schedulable until PVC is created", "pod", klog.KObj(pod), "err", err)
+				return framework.QueueSkip, nil
+			}
+			return framework.Queue, err
+		}
+
+		if v1helper.ContainsAccessMode(pvc.Spec.AccessModes, v1.ReadWriteOncePod) {
+			readWriteOncePodPVC.Insert(pvc.Name)
+		}
+
+		pvcs.Insert(pvc.Name)
+	}
+
+	if oldPersistentVolumeClaim == nil {
+		// If the added pvc is not associated with a pod,
+		// return QueueSkip to prevent the pod from being queued when any PVCs are added,
+		// we only queue the pod when we create that pod's pvc.
+		if !pvcs.Has(newPersistentVolumeClaim.Name) {
+			return framework.QueueSkip, nil
+		}
+		// We need to queue the pod when we create the pvc for the pod.
+		return framework.Queue, nil
```

Comment:
```suggestion
		// But, we cannot return Queue here because we have to check readWriteOncePodPVCs as well.
```

Reply:
We should return `framework.Queue`, but we should first check if `newPersistentVolumeClaim` conflicts with an existing pvc

If a pvc is created, then when it does not conflict with other `readWriteOncePodPVC`, we should return `framework.Queue`, because it almost proves that the pod can be scheduled; if it conflicts with other `readWriteOncePodPVC`, we should return `framework.QueueSkip` , waiting for the deletion events of other pods and `readWriteOncePodPVC` to be scheduled again

I removed the judgment of whether all pvcs of the current pod conflict with the existing `readWriteOncePodPVC`, because if they conflict, they need to return `framework.QueueSkip` and wait for the deletion events of other pods and `readWriteOncePodPVC` to be scheduled again, which is the same as the default return value of the function.

What do you think?